### PR TITLE
Feature/last succesfull upload

### PR DESCRIPTION
**Pull request by prinsniels**
First iteration

### CODE REVIEW

File: app/src/main/java/nl/nfi/cellscanner/ExportResultRepository.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    public static boolean getLastUploadSuccessful(Context ctx) {
+        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(ctx);
+        return prefs.getBoolean(LAST_UPLOAD_SUCCESSFUL, false);
+    }
+
+    public static String getLastUploadType(Context ctx) {
+        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(ctx);
+        return prefs.getString(LAST_UPLOAD_TYPE, "UNKNOWN_ERROR");
```
Comment (wowtor) [inline app/src/main/java/nl/nfi/cellscanner/ExportResultRepository.java]: Is it an unknown error, or unknown status?

File: app/src/main/java/nl/nfi/cellscanner/ViewMeasurementsActivity.java [hunk 2]
```diff
@@ -145,5 +131,23 @@ private static String getDateTimeFromTimeStamp(Long time, String requestedDateFo
         Date dateTime = new Date(time);
         return dateFormat.format(dateTime);
     }
+
+    private void setAutoUploadData() {
+        vlUpStatus.setText(ExportResultRepository.getLastUploadMsg(getApplicationContext()));
+        vlUpLastSuccess.setText(getDateTimeFromTimeStamp(ExportResultRepository.getLastSuccessfulUploadTimestamp(getApplicationContext()), "yyyy-MM-dd HH:mm:ss"));
+        vlUpLastUpdate.setText(getDateTimeFromTimeStamp(ExportResultRepository.getLastUploadTimeStamp(getApplicationContext()),"yyyy-MM-dd HH:mm:ss"));
```
Comment (wowtor) [inline app/src/main/java/nl/nfi/cellscanner/ViewMeasurementsActivity.java:138]: timestamp may be `0` -> "never"

File: app/src/main/java/nl/nfi/cellscanner/PreferencesActivity.java [hunk 3]
```diff
@@ -247,6 +247,7 @@ private void updateButtonStateToRecordingState() {
     @Override
     public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {
         // TODO, this should be removed, as the switches are not externally controlled
+        Log.i("WORKING", key);
```
Comment (wowtor) [inline app/src/main/java/nl/nfi/cellscanner/PreferencesActivity.java]: why this message?

File: app/src/main/java/nl/nfi/cellscanner/UserDataUploadWorker.java [hunk 4]
```diff
@@ -65,12 +63,16 @@ public Result doWork() {
...
+                    // SIGNAL our success
                     Log.i(TAG, "upload result: succeeded");
-
+                    ExportResultRepository.storeExportResult(getApplicationContext(), timestamp, true, "success", ExportResultRepository.AUTO);
                     db.dropDataUntil(timestamp);
 
                 } else {
+                    // SIGNAL their failure
                     Log.i(TAG, "upload result: Failed");
+                    ExportResultRepository.storeExportResult(getApplicationContext(), timestamp, false, "unknown failure", ExportResultRepository.AUTO);
```
Comment (wowtor) [inline app/src/main/java/nl/nfi/cellscanner/UserDataUploadWorker.java]: unknown failure -> upload failed

File: app/src/main/java/nl/nfi/cellscanner/ViewMeasurementsActivity.java [hunk 5]
```diff
@@ -145,5 +131,23 @@ private static String getDateTimeFromTimeStamp(Long time, String requestedDateFo
...
+    }
+
+    @Override
+    public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {
+        /*
+        The last automatic upload data is stored as a shared preference.
+        When this data changes, the data is retrieved and placed
+        on the screen
+         */
+        Log.i("WORK", key);
```
Comment (wowtor) [inline app/src/main/java/nl/nfi/cellscanner/ViewMeasurementsActivity.java:148]: work? if we need this, verbosity should probably be debug

File: app/src/main/java/nl/nfi/cellscanner/PreferencesActivity.java (hunk 3 above)
Comment (prinsniels) [reply app/src/main/java/nl/nfi/cellscanner/PreferencesActivity.java]: debug should be removed

**Pull request merged**